Route/RoutePageGenerator.php

```diff
@@ -99,9 +99,20 @@ public function update(SiteInterface $site, OutputInterface $output = null)
         }
 
         // Iterate over declared routes from the routing mechanism
+        $routesDedupInfos = array();
+        // We first de-duplicate routes by path; the first declaration win !
         foreach ($this->router->getRouteCollection()->all() as $name => $route) {
-            $name = trim($name);
+            $routePath = $route->getPath();
+            if (array_key_exists($routePath, $routesDedupInfos)) {
+                $this->writeln($output, sprintf('  <info>IGNORE</info> route %s is ignored because it uses the same path as %s (path=%s)',
+                    $name, $routesDedupInfos[$routePath][0], $routePath));
```

Comment:
I do it regularly but I hardly can do it for the whole codebase, sorry. I'll do it for the examples you pointed out though. Some of the code is really old and there are not fixers for everything.